lib/http/response.rb

```diff
@@ -156,8 +156,8 @@ def chunked?
     # @param type [#to_s] Parse as given MIME type.
     # @raise (see MimeType.[])
     # @return [Object]
-    def parse(type)
-      MimeType[type].decode to_s
+    def parse(type = nil)
+      MimeType[type || mime_type].decode to_s
```

Comment:
How about making default to be mime-type:

```suggestion
    def parse(as = mime_type)
      MimeType[as].decode to_s
```

Reply:
/cc @tarcieri @DannyBen

Reply:
I find this pattern to be bad, but don't have strong feelings either way :D